optimum/neuron/modeling_base.py

```diff
@@ -0,0 +1,440 @@
+    """
+
+    model_type = "neuron_model"
+    auto_model_class = AutoModel
+
+    def __init__(
+        self,
+        model: torch.jit._script.ScriptModule,
+        config: "PretrainedConfig",
+        model_save_dir: Optional[Union[str, Path, TemporaryDirectory]] = None,
+        model_file_name: Optional[str] = None,
+        preprocessors: Optional[List] = None,
+        neuron_config: "NeuronConfig" = None,
+        **kwargs,
+    ):
+        super().__init__(model, config)
+
+        self.model = model
+        self.model_file_name = model_file_name or NEURON_FILE_NAME
+        self.config = config
+        self.neuron_config = self._neuron_config_init(self.config) if neuron_config is None else neuron_config
+        self.input_static_shapes = NeuronModel.get_input_static_shapes(self.neuron_config)
+        self._attributes_init(model_save_dir, preprocessors, **kwargs)
+
+    @staticmethod
+    def load_model(path: Union[str, Path]) -> torch.jit._script.ScriptModule:
+        """
+        Loads a TorchScript module compiled by neuron(x)-cc compiler. It will be first loaded onto CPU and then moved to
+        one or multiple [NeuronCore](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/general/arch/neuron-hardware/neuroncores-arch.html).
+
+        Args:
+            path (`Union[str, Path]`):
+                Path of the compiled model.
+        """
+        if not isinstance(path, str):
+            path = str(path)
+
+        return torch.jit.load(path)
+
+    def _save_pretrained(self, save_directory: Union[str, Path]):
+        """
+        Saves a model and its configuration file to a directory, so that it can be re-loaded using the
+        [`~optimum.neuron.modeling_base.NeuronModel.from_pretrained`] class method.
+
+        Args:
+            save_directory (`Union[str, Path]`):
+                Directory where to save the model file.
+        """
+        src_paths = [self.model_save_dir / self.model_file_name]
+        dst_paths = [Path(save_directory) / self.model_file_name]
+
+        for src_path, dst_path in zip(src_paths, dst_paths):
+            shutil.copyfile(src_path, dst_path)
+
+    @classmethod
+    def _from_pretrained(
+        cls,
+        model_id: Union[str, Path],
+        config: "PretrainedConfig",
+        use_auth_token: Optional[Union[bool, str]] = None,
+        revision: Optional[str] = None,
+        force_download: bool = False,
+        cache_dir: Optional[str] = None,
+        file_name: Optional[str] = None,
+        subfolder: str = "",
+        local_files_only: bool = False,
+        model_save_dir: Optional[Union[str, Path, TemporaryDirectory]] = None,
+        neuron_config: "NeuronConfig" = None,
+        **kwargs,
+    ) -> "NeuronModel":
+        model_path = Path(model_id)
+
+        if file_name is None:
+            if model_path.is_dir():
+                neuron_files = list(model_path.glob("*.neuron"))
+            else:
+                if isinstance(use_auth_token, bool):
```

Comment:
```suggestion
                if isinstance(use_auth_token, bool) and use_auth_token:
```

Reply:
What if `use_auth_token` is left to None?

Reply:
Then the `token = use_auth_token = None` so should be fine.